### PR TITLE
[quant][graphmode] Add quantize_per_tensor.tensors

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#35916 [quant][graphmode] Add quantize_per_tensor.tensors**
* #35894 [quant][graphmode] Insert Observers for dynamic LSTM
* #35893 [quant][graphmode] Add new tensorlist observer for LSTM

Summary:
quantize_per_tensor can now accept list of tensors.
Needed for operators like LSTM and cat

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20830388](https://our.internmc.facebook.com/intern/diff/D20830388)

### CODE REVIEW

File: aten/src/ATen/native/quantized/QTensor.cpp [hunk 1]
```diff
@@ -18,6 +18,22 @@ Tensor quantize_per_tensor_cpu(
   return quantizer->quantize(self);
 }
 
+std::vector<Tensor> quantize_per_tensor_list_cpu(
```
Comment (supriyar) [inline aten/src/ATen/native/quantized/QTensor.cpp:21]: @jerryzh168 just confirming that this is the right place to define it? I seem to be hitting linker errors.

Comment (jerryzh168) [reply aten/src/ATen/native/quantized/QTensor.cpp:21]: I think so

File: aten/src/ATen/native/quantized/QTensor.cpp [hunk 2]
```diff
@@ -18,6 +18,22 @@ Tensor quantize_per_tensor_cpu(
...
+    TensorList tensors,
+    Tensor scales,
+    Tensor zero_points,
+    ScalarType dtype) {
+  std::vector<Tensor> quantized_tensors;
+  for (auto i = 0; i < tensors.size(); ++i) {
+    quantized_tensors.push_back(quantize_per_tensor_cpu(
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/QTensor.cpp:28]: should we call aten native function here?

Comment (supriyar) [reply aten/src/ATen/native/quantized/QTensor.cpp:28]: Either way works I think. That function calls `quantize_per_tensor_cpu`

**Comment by jerryzh168** [comment]
you might need to do similar changes to the python lint files as I did in https://github.com/pytorch/pytorch/pull/34348

**Review by jerryzh168 — APPROVED** [review]
Please wait for CI, I think this will break the type_hint tests

**Pull request closed without merging**